Modules/Devices/test/Devices.Application.Tests/Tests/Identities/Commands/CancelDeletionProcess/HandlerTests.cs

```diff
@@ -21,47 +18,58 @@ public async Task Happy_path()
         // Arrange
         var activeIdentity = TestDataGenerator.CreateIdentityWithApprovedDeletionProcess(SystemTime.UtcNow.Date);
         var activeDevice = activeIdentity.Devices[0];
+        var deletionProcess = activeIdentity.GetDeletionProcessInStatus(DeletionProcessStatus.Approved)!;
 
-        var mockIdentitiesRepository = A.Fake<IIdentitiesRepository>();
+        var fakeIdentitiesRepository = A.Fake<IIdentitiesRepository>();
         var fakeUserContext = A.Fake<IUserContext>();
 
-        A.CallTo(() => mockIdentitiesRepository.FindByAddress(activeIdentity.Address, CancellationToken.None, A<bool>._))
+        A.CallTo(() => fakeIdentitiesRepository.FindByAddress(activeIdentity.Address, CancellationToken.None, A<bool>._))
             .Returns(activeIdentity);
         A.CallTo(() => fakeUserContext.GetAddress()).Returns(activeIdentity.Address);
         A.CallTo(() => fakeUserContext.GetDeviceId()).Returns(activeDevice.Id);
 
-        var handler = new Handler(mockIdentitiesRepository, fakeUserContext);
-        var command = new CancelDeletionProcessCommand(activeDevice.Id);
+        var handler = CreateHandler(fakeIdentitiesRepository, fakeUserContext);
+        var command = new CancelDeletionProcessCommand(deletionProcess.Id);
 
         // Act
-        await handler.Handle(command, CancellationToken.None);
+        var response = await handler.Handle(command, CancellationToken.None);
 
         // Assert
-        var result = activeIdentity.GetDeletionProcessInStatus(DeletionProcessStatus.Cancelled)!;
-        result.Status.Should().Be(DeletionProcessStatus.Cancelled);
+        A.CallTo(() => fakeIdentitiesRepository.Update(A<Identity>.That.Matches(i =>
+                i.Address == activeIdentity.Address
+                && i.Status == IdentityStatus.Active
+                && i.DeletionProcesses.Any(d => d.Id == deletionProcess.Id)), A<CancellationToken>._))
+            .MustHaveHappenedOnceExactly();
+
+        response.Status.Should().Be(DeletionProcessStatus.Cancelled);
     }
 
     [Fact]
     public void Cannot_start_when_given_identity_does_not_exist()
     {
         // Arrange
         var address = CreateRandomIdentityAddress();
+        var dummyUserContext = A.Fake<IUserContext>();
```

Comment:
If you are not setting up anything special with this user context then it should be created like this in the private `CreateHandler` method.